src/sql/src/plan/query.rs

```diff
@@ -5586,15 +5590,50 @@ impl Visit<'_, Aug> for WindowFuncCollector {
     }
 }
 
-/// Specifies how long a query will live. This impacts whether the query is
+/// Specifies how long a query will live.
+///
+/// Currently, it affects whether SHOW commands are allowed.
+///
+/// (This used to also impact whether the query is
 /// allowed to reason about the time at which it is running, e.g., by calling
-/// the `now()` function.
+/// the `now()` function. Nowadays, this is decided by a different mechanism.)
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 pub enum QueryLifetime {
-    /// The query's result will be computed at one point in time.
+    /// The query's (or the expression's) result will be computed at one point in time.
     OneShot,
```

Comment:
In line with the remaining variants in this enum I suggest at some point to rename `OneShot` to `Select`, but there might be reasons not to do this.

Reply:
Well, saying "select" for one-shot queries sounds a bit confusing to me, as non-one-shot queries also involve `SELECT`. So I'd rather say "one-shot" where possible.